R/mlr_graphs_elements.R

```diff
@@ -243,3 +244,67 @@ pipeline_branch = function(graphs, prefix_branchops = "", prefix_paths = FALSE)
 }
 
 mlr_graphs$add("branch", pipeline_branch)
+
+
+
+#' @title Transform A Numerical Target And Later Invert The Transformation
+#' @name mlr_graphs_targettrafo
+#' @description
+#' Creates a [`Graph`] that transforms a numerical target of a [`TaskRegr`][mlr3::TaskRegr] during
+#' training and inverts the transformation during prediction.
+#' This is done as follows:
+#' * Specify a transformation and inversion function using [`PipeOpTargetTrafoSimple`], afterwards
+#'   apply `graph`.
+#' * At the very end, during prediction the transformation is inverted using [`PipeOpTargetInverter`].
+#' * To set a transformation and inversion function see the parameters `targettrafosimple.trafo` and
+#'   `targettrafosimple.inverter` of the `param_set` of the resulting [`Graph`].
+#'
+#' @param graph [`PipeOpLearner`] | [`Graph`] \cr
+#'   A [`PipeOpLearner`] or [`Graph`] to create a robustifying pipeline for. If this is a [`Graph`], the
+#'   last [`PipeOp`] should be a [`PipeOpLearner`] that can handle a [`TaskRegr`][mlr3::TaskRegr] and
+#'   the first [`PipeOp`] should accept a single [`TaskRegr`][mlr3::TaskRegr] as input.
+#'
+#' @return [`Graph`]
+#' @export
+#' @examples
+#'library("mlr3")
+#'
+#'targettrafo = pipeline_targettrafo(PipeOpLearner$new(LearnerRegrRpart$new()))
+#'targettrafo$param_set$values$targettrafosimple.trafo = function(x) log(x, base = 2)
+#'targettrafo$param_set$values$targettrafosimple.inverter = function(x) 2 ^ x
+#'
+#'# gives the same as
+#'g = Graph$new()
+#'g$add_pipeop(PipeOpTargetTrafoSimple$new(param_vals = list(
+#'  trafo = function(x) log(x, base = 2),
+#'  inverter = function(x) 2 ^ x)
+#'  )
+#')
+#'g$add_pipeop(LearnerRegrRpart$new())
+#'g$add_pipeop(PipeOpTargetInverter$new())
+#'g$add_edge(src_id = "targettrafosimple", dst_id = "targetinverter",
+#'  src_channel = 1, dst_channel = 1)
+#'g$add_edge(src_id = "targettrafosimple", dst_id = "regr.rpart",
+#'  src_channel = 2, dst_channel = 1)
+#'g$add_edge(src_id = "regr.rpart", dst_id = "targetinverter",
+#'  src_channel = 1, dst_channel = 2)
+pipeline_targettrafo = function(graph) {
```

Comment:
signature should be `function(graph, trafo_pipeop = NULL, id_prefix = "")`
If  `trafo_pipeop = NULL` uses `PipeOpTargetTrafoSimple`, otherwise a custom target trafo can be provided.
Additionally prefixes the `id` of trafo and inverter with `id_prefix`. 
Otherwise, this could lead to issues as duplicate id's are not allowed.

Test that this works for custom id's aswell.